src/compiler/output-targets/dist-custom-elements/test/dist-custom-elements.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { addCustomElementInputs } from '../index';
```

Comment:
I'd like to add a bit more type safety to these unit tests if possible. Stencil has a lot of interfaces/types that have _several_ properties on them, making them a bit cumbersome to build out for each and every test suite (or even on a test-by-test basis).

We have a handful of stubs/mock helper functions to provide us with a "known good" baseline for some of these types, which we can override at construction time (most of the time). The API of these mocks/stubs is a little inconsistent due to the project changing hands over the years, but I _think_ we have a few that could be of use here. Take a look at the included patch, and let me know what you think - I'll add comments here in the PR as well, but they'll be a bit fragmented (hence including that patch).

[addtl-type-safety.patch](https://github.com/ionic-team/stencil/files/14434817/addtl-type-safety.patch)

Reply:
This is amazing, thank you!